src/api/chat-api.ts

```diff
@@ -0,0 +1,29 @@
+import { BaseUrl } from "../utils/const";
+import { HTTPTransport } from "../utils/http-transport";
+import { IChat, IUserInChats } from "../utils/types";
+
+const settingsAPIInstance = new HTTPTransport(BaseUrl);
+
+export class ChatAPI {
+  public getChats() {
+    return settingsAPIInstance.get("/chats");
+  }
```

Comment:
Вот тут неочевидная, но важная деталь есть (в этом, и в других методах остальных API классах)

В методе класса используется объект, который находится вне этого класса. Это создает жесткую зависимость и нарушает инкапсуляцию. Было бы правильнее сделать свойство класса, инициализировать его в конструктора (или передавать через параметры конструктора) и в методах этого класса ссылкаться на это свойство. Например так
```suggestion
  private settingsAPIInstance: HTTPTransport;
  
  constructor() {
   this.settingsAPIInstance = new HTTPTransport(BaseUrl)
  }

  public getChats() {
    return this.settingsAPIInstance.get("/chats");
  }
```